sdk/src/main/java/io/opentdf/platform/sdk/TDF.java

```diff
@@ -323,8 +318,56 @@ private void prepareManifest(Config.TDFConfig tdfConfig, SDK.KAS kas) {
 
             this.aesGcm = new AesGcm(this.payloadKey);
         }
+
+        private Manifest.KeyAccess createKeyAccess(Config.TDFConfig tdfConfig, Config.KASInfo kasInfo, byte[] symKey, Manifest.PolicyBinding policyBinding, String encryptedMetadata, String splitID) {
+            Manifest.KeyAccess keyAccess = new Manifest.KeyAccess();
+            keyAccess.keyType = kWrapped;
+            keyAccess.url = kasInfo.URL;
+            keyAccess.kid = kasInfo.KID;
+            keyAccess.protocol = kKasProtocol;
+            keyAccess.policyBinding = policyBinding;
+            keyAccess.encryptedMetadata = encryptedMetadata;
+            keyAccess.sid = splitID;
+            keyAccess.schemaVersion = KEY_ACCESS_SECHMA_VERSION;
+
+            if (tdfConfig.wrappingKeyType != KeyType.RSA2048Key) {
+                var ecKeyWrappedKeyInfo =createECWrappedKey(tdfConfig, kasInfo, symKey);
+                keyAccess.wrappedKey = ecKeyWrappedKeyInfo.wrappedKey;
+                keyAccess.ephemeralPublicKey = ecKeyWrappedKeyInfo.publicKey;
+                keyAccess.keyType = kECWrapped;
+            } else {
+                keyAccess.wrappedKey = createRSAWrappedKey(kasInfo, symKey);
+                keyAccess.keyType = kWrapped;
+            }
+            return keyAccess;
+        }
+
+        private ECKeyWrappedKeyInfo createECWrappedKey(Config.TDFConfig tdfConfig, Config.KASInfo kasInfo, byte[] symKey)  {
+            var curveName = tdfConfig.wrappingKeyType.getCurveName();
+            var keyPair = new ECKeyPair(curveName, ECKeyPair.ECAlgorithm.ECDH);
+
+            ECPublicKey kasPubKey = ECKeyPair.publicKeyFromPem(kasInfo.PublicKey);
+            byte[] symmetricKey = ECKeyPair.computeECDHKey(kasPubKey, keyPair.getPrivateKey());
+
+            var sessionKey = ECKeyPair.calculateHKDF(GLOBAL_KEY_SALT, symmetricKey);
+
+            AesGcm gcm = new AesGcm(sessionKey);
+            AesGcm.Encrypted wrappedKey = gcm.encrypt(symKey);
+
+            ECKeyWrappedKeyInfo wrappedKeyInfo = new ECKeyWrappedKeyInfo();
+            wrappedKeyInfo.publicKey = keyPair.publicKeyInPEMFormat();
+            wrappedKeyInfo.wrappedKey = Base64.getEncoder().encodeToString(wrappedKey.asBytes());
+            return wrappedKeyInfo;
```

Comment:
Same thing here. This could all go inside an `AsymEncrypt`. This is fine, though.